pandas/core/reshape/pivot.py

```diff
@@ -454,6 +454,8 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
 
     from pandas import DataFrame
     df = DataFrame(data, index=common_idx)
+    if not len(df):
```

Comment:
rather than do this (which is incorrect because the result doesn't have the correct index).

just change next statement to 

```
if len(df) and values is None:
....
```

Reply:
I think I tried that and got an assertion error because the test was using a new empty dataframe as expected value, and the return value was an empty dataframe with a different index. I'll check again, and if it still breaks I'll change the expected value accordingly, assuming no other tests break.

Reply:
Changing the statement to `if len(df) and values is None:` stills makes the crosstab raise the ValueError I'm implementing in this PR, because in this test len(df) is 0 and values IS None, so when it falls to:
``` python
else:
        df['__dummy__'] = values
        kwargs = {'aggfunc': aggfunc}
```
it's basically doing  `df['__dummy__'] = None` which is what raises the error.

What I think would fix it and preserve the index would be this:
``` python
if not len(df):
    return df
```

Reply:
see my comment below.

Reply:
actually just return did here i think is ok (and u had that before)

change and ping on green

Reply:
I'm sorry, I'm not sure I understand what you mean. Should I change it back to return an empty DataFrame?

Reply:
no change back to 

```
if not len(df)
    return df
```
you need to return the DataFrame with the correct index.

Reply:
Ok, changed it back :)